lineoperations/src/lineoperations.c

```diff
@@ -0,0 +1,203 @@
+	rmdupln_item  = gtk_menu_item_new_with_mnemonic(_("Remove Duplicate Lines, _Ordered"));
+	rmunqln_item  = gtk_menu_item_new_with_mnemonic(_("Remove _Unique Lines"));
+	rmemtyln_item = gtk_menu_item_new_with_mnemonic(_("Remove _Empty Lines"));
+	rmwhspln_item = gtk_menu_item_new_with_mnemonic(_("Remove _Whitespace Lines"));
+	sortasc_item  = gtk_menu_item_new_with_mnemonic(_("Sort Lines _Ascending"));
+	sortdesc_item = gtk_menu_item_new_with_mnemonic(_("Sort Lines _Descending"));
+
+	gtk_widget_show(sep1);
+	gtk_widget_show(sep2);
+	gtk_widget_show(rmdupst_item);
+	gtk_widget_show(rmdupln_item);
+	gtk_widget_show(rmunqln_item);
+	gtk_widget_show(rmemtyln_item);
+	gtk_widget_show(rmwhspln_item);
+	gtk_widget_show(sortasc_item);
+	gtk_widget_show(sortdesc_item);
+
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), rmdupst_item);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), rmdupln_item);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), rmunqln_item);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), sep1);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), rmemtyln_item);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), rmwhspln_item);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), sep2);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), sortasc_item);
+	gtk_menu_shell_append(GTK_MENU_SHELL(submenu), sortdesc_item);
+
+	gtk_menu_item_set_submenu(GTK_MENU_ITEM(main_menu_item), submenu);
+
+	gtk_container_add(GTK_CONTAINER(geany->main_widgets->tools_menu), main_menu_item);
+
+
+	g_signal_connect(rmdupst_item, "activate", G_CALLBACK(action_rmdupst_item), NULL);
+	g_signal_connect(rmdupln_item, "activate", G_CALLBACK(action_rmdupln_item), NULL);
+	g_signal_connect(rmunqln_item, "activate", G_CALLBACK(action_rmunqln_item), NULL);
+	g_signal_connect(rmemtyln_item, "activate", G_CALLBACK(action_rmemtyln_item), NULL);
+	g_signal_connect(rmwhspln_item, "activate", G_CALLBACK(action_rmwhspln_item), NULL);
+	g_signal_connect(sortasc_item, "activate", G_CALLBACK(action_sortasc_item), NULL);
+	g_signal_connect(sortdesc_item, "activate", G_CALLBACK(action_sortdesc_item), NULL);
+
+	ui_add_document_sensitive(rmdupst_item);
+	ui_add_document_sensitive(rmdupln_item);
+	ui_add_document_sensitive(rmunqln_item);
+	ui_add_document_sensitive(rmemtyln_item);
+	ui_add_document_sensitive(rmwhspln_item);
+	ui_add_document_sensitive(sortasc_item);
+	ui_add_document_sensitive(sortdesc_item);
+
+	return TRUE;
+}
+
+
+void
+lo_cleanup(void)
+{
+	if(main_menu_item) gtk_widget_destroy(main_menu_item);
+}
+
+
+G_MODULE_EXPORT
+void geany_load_module(GeanyPlugin *plugin)
+{
+    plugin->info->name        = _("Line Operations");
+    plugin->info->description = _("Line Operations provides a handful of functions that can be applied to a document such as, removing duplicate lines, removing empty lines, removing lines with only whitespace, and sorting lines.");
+    plugin->info->version     = "0.1";
+    plugin->info->author      = _("Sylvan Mostert <smostert.dev@gmail.com>");
```

Comment:
To allow those strings actually be displayed translated, you should call `main_locale_init()` before filling the plugin info (well, before any call to `_()`).  See http://www.geany.org/manual/reference/howto.html#translatable_plugin_information

Note: `main_loacle_init()` only has to be called once per plugin run, so calling it here will be enough for the whole plugin, no need to call it again in `init()` or anything.